### ISSUE TITLE
allow for verifying a mar with an unknown signature

### ISSUE THREAD

**Issue by escapewindow** [mozilla/build-mar#45]
Currently, we can check if a mar is either good and has a valid signature, or is bad or has an invalid signature. I'd like a check to make sure a mar file is good (the mar itself is well-formed) and the signature is well-formed, even if the signature is from an unknown key. This could help when testing signing against throwaway keys; we may encounter this scenario when making sure a new signing pipeline is working, since we don't always want to use shipping keys for that type of testing.

I was able to work around this limitation with a couple print calls:

```diff
diff --git a/src/mardor/reader.py b/src/mardor/reader.py
index fb90afe..179d847 100644
--- a/src/mardor/reader.py
+++ b/src/mardor/reader.py
@@ -160,6 +160,7 @@ class MarReader(object):
         """
         if not self.mardata.signatures or not self.mardata.signatures.sigs:
             # This MAR file can't be verified since it has no signatures
+            print("No sigs")
             return False

         verifiers = []
@@ -183,6 +184,7 @@ class MarReader(object):
             try:
                 v.verify()
             except InvalidSignature:
+                print("Sig doesn't match key")
                 return False
         else:
             return True
```

**Comment by catlee** [comment]
On the command line, we could make the -v option check that the mar is well-formed, and if -k is passed it additionally will verify the signature. Would that work?

**Comment by escapewindow** [comment]
Sounds good to me!

**Comment by catlee** [comment]
fixed by 6cba2bcaaaa214776349b8e040f998724b47684c in v3.0.0
